<a href="https://colab.research.google.com/github/Srikhanth14/Data_Analysis_Projects/blob/main/HealthCare_Resource_Optimization.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Introduction**

This module focuses on developing a machine learning model to predict the historical admission type of a patient using demographic, medical, hospital, insurance and temporal information. Multiple classification algorithms are evaluated and compared using accuracy, precision, recall and F1-score. The best-performing model is subsequently selected for integration into the healthcare resource optimization system.



# **Import the Necessary Libraries**

In [1]:
import pandas as pd
import numpy as np

# **Data Collection and Processing**

In [2]:
#Loading the Datasets
data = pd.read_csv("modified_healthcare_dataset.csv")

In [3]:
# Display first three rows of the dataset
data.head(3)

,Name,Age,Gender,Blood Type,Medical Condition,Date of Admission,Doctor,Hospital,Insurance Provider,Billing Amount,Room Number,Admission Type,Discharge Date,Medication,Test Results,Length of Stay
0,Bobby Jackson,19,Female,AB+,Infections,2024-01-31,Matthew Smith,Northwestern Memorial Hospital,Blue Cross,2212.272701,328,Emergency,2024-02-07,Azithromycin,Normal,7
1,Leslie Terry,15,Female,B-,Flu,2019-08-20,Samantha Davies,UI Health (University of Illinois Hospital),UnitedHealthcare,3185.161388,265,Emergency,2019-08-22,Tamiflu,Abnormal,2
2,Danny Smith,50,Female,A+,Cancer,2022-09-22,Tiffany Mitchell,UI Health (University of Illinois Hospital),Blue Cross,72055.214065,205,Elective,2022-10-30,Cisplatin,Inconclusive,38


In [4]:
# Display last four rows of the dataset
data.tail(4)

,Name,Age,Gender,Blood Type,Medical Condition,Date of Admission,Doctor,Hospital,Insurance Provider,Billing Amount,Room Number,Admission Type,Discharge Date,Medication,Test Results,Length of Stay
55496,Kyle Perez,63,Male,A+,Heart Disease,2020-01-23,Taylor Sullivan,UI Health (University of Illinois Hospital),Medicare,26828.223456,316,Emergency,2020-03-07,Statins,Abnormal,44
55497,Heather Wang,28,Female,O+,Asthma,2020-07-13,Joe Jacobs DVM,Northwestern Memorial Hospital,Cigna,6041.030866,347,Elective,2020-07-16,Montelukast,Normal,3
55498,Jennifer Jones,75,Male,O+,Cancer,2019-05-25,Kimberly Curry,UChicago Medicine,Aetna,44495.656362,321,Emergency,2019-07-11,Methotrexate,Abnormal,47
55499,James Garcia,53,Female,O+,Diabetes,2024-04-02,Dennis Warren,Loyola University Medical Center,Cigna,15953.455966,448,Emergency,2024-04-12,Insulin,Abnormal,10


In [5]:
#Display the Sample of any 2 rows from the dataset
data.sample(2)

,Name,Age,Gender,Blood Type,Medical Condition,Date of Admission,Doctor,Hospital,Insurance Provider,Billing Amount,Room Number,Admission Type,Discharge Date,Medication,Test Results,Length of Stay
35063,Matthew Davis,72,Female,A+,Alzheimer’s,2019-12-22,Earl Jimenez,UChicago Medicine,Medicare,21797.642936,292,Emergency,2020-01-16,Donepezil,Inconclusive,25
6874,Charles Barker,65,Female,B+,Alzheimer’s,2022-12-09,Dr. David Evans,Northwestern Memorial Hospital,Medicare,42041.261580,152,Emergency,2023-02-05,Donepezil,Abnormal,58


In [6]:
#Print the Dimensions of the dataset
print("Number of Rows: ", data.shape[0])
print("Number of Columns: ", data.shape[1])

Number of Rows:  55500
Number of Columns:  16


In [7]:
#Displaying all column names in the Dataset
for col in data.columns:
  print(col)

Name
Age
Gender
Blood Type
Medical Condition
Date of Admission
Doctor
Hospital
Insurance Provider
Billing Amount
Room Number
Admission Type
Discharge Date
Medication
Test Results
Length of Stay


# **Findings**
The dataset contains 55,500 patient records and 16 attributes. The dataset includes demographic information, medical conditions, hospital details, admission and discharge dates, billing information, medication, test results and length of stay

# **Data Quality Analysis**

In [8]:
#Checking the Data type of Columns in the Dataset
data.dtypes.to_frame(name="Data_Type")

,Data_Type
Name,object
Age,int64
Gender,object
Blood Type,object
Medical Condition,object
Date of Admission,object
Doctor,object
Hospital,object
Insurance Provider,object
Billing Amount,float64


In [9]:
# Checking any null values present in the each column
if data.isnull().sum().sum() == 0:
  print("No null values present in the dataset")
else:
  print("Null values present in the dataset")

No null values present in the dataset


In [10]:
print(data.isnull().sum())

Name                  0
Age                   0
Gender                0
Blood Type            0
Medical Condition     0
Date of Admission     0
Doctor                0
Hospital              0
Insurance Provider    0
Billing Amount        0
Room Number           0
Admission Type        0
Discharge Date        0
Medication            0
Test Results          0
Length of Stay        0
dtype: int64


In [11]:
#Checking any duplicate values present in the Dataset
if data.duplicated().sum() == 0:
  print("Duplicate values doesn't exist in the dataset")
else:
  print("Duplicate values exist in the dataset")

Duplicate values doesn't exist in the dataset


In [12]:
#Statistical Summary of the Dataset
data.describe()

,Age,Billing Amount,Room Number,Length of Stay
count,55500.000000,55500.000000,55500.000000,55500.000000
mean,48.001622,21835.044635,301.134829,17.841009
std,21.105827,23574.413594,115.243069,20.272294
min,5.000000,500.220989,101.000000,1.000000
25%,32.000000,4195.525741,202.000000,4.000000
50%,50.000000,11089.537763,302.000000,8.000000
75%,65.000000,35887.703406,401.000000,28.000000
max,90.000000,99997.797977,500.000000,89.000000


In [13]:
#Overview of the Dataset
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 55500 entries, 0 to 55499
Data columns (total 16 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Name                55500 non-null  object 
 1   Age                 55500 non-null  int64  
 2   Gender              55500 non-null  object 
 3   Blood Type          55500 non-null  object 
 4   Medical Condition   55500 non-null  object 
 5   Date of Admission   55500 non-null  object 
 6   Doctor              55500 non-null  object 
 7   Hospital            55500 non-null  object 
 8   Insurance Provider  55500 non-null  object 
 9   Billing Amount      55500 non-null  float64
 10  Room Number         55500 non-null  int64  
 11  Admission Type      55500 non-null  object 
 12  Discharge Date      55500 non-null  object 
 13  Medication          55500 non-null  object 
 14  Test Results        55500 non-null  object 
 15  Length of Stay      55500 non-null  int64  
dtypes: f

In [14]:
#Identifying the Total unique values in Each Column
data.nunique()

,0
Name,40235
Age,86
Gender,2
Blood Type,8
Medical Condition,8
Date of Admission,1827
Doctor,40341
Hospital,4
Insurance Provider,5
Billing Amount,55500


In [15]:
print("Distribution of Each Categorical Variables",'\n')
Categorical_Target_Columns = ["Gender","Blood Type", "Medical Condition", "Doctor", "Hospital", "Insurance Provider", "Admission Type", "Medication", "Test Results"]

for categorical_columns in Categorical_Target_Columns:
    print(data[categorical_columns].value_counts())
    print("-"*50)

Distribution of Each Categorical Variables 

Gender
Female    27955
Male      27545
Name: count, dtype: int64
--------------------------------------------------
Blood Type
O+     15696
A+     15573
B+      5457
AB-     4485
B-      4408
O-      3341
A-      3301
AB+     3239
Name: count, dtype: int64
--------------------------------------------------
Medical Condition
Flu              7046
Diabetes         7005
Obesity          6994
Cancer           6940
Asthma           6908
Heart Disease    6900
Alzheimer’s      6861
Infections       6846
Name: count, dtype: int64
--------------------------------------------------
Doctor
Michael Smith      27
John Smith         22
Robert Smith       22
Michael Johnson    20
James Smith        20
                   ..
Keith Ortiz         1
Timothy Chapman     1
Terri Collins       1
Jill Jacobson       1
Scott Coleman       1
Name: count, Length: 40341, dtype: int64
--------------------------------------------------
Hospital
Northwestern Memorial Hosp

In [16]:
print("Distribution of Categorical Variables in Percentage",'\n')
for categorical_columns in Categorical_Target_Columns:
    print(data[categorical_columns].value_counts(normalize=True)*100)
    print("-"*50)

Distribution of Categorical Variables in Percentage 

Gender
Female    50.369369
Male      49.630631
Name: proportion, dtype: float64
--------------------------------------------------
Blood Type
O+     28.281081
A+     28.059459
B+      9.832432
AB-     8.081081
B-      7.942342
O-      6.019820
A-      5.947748
AB+     5.836036
Name: proportion, dtype: float64
--------------------------------------------------
Medical Condition
Flu              12.695495
Diabetes         12.621622
Obesity          12.601802
Cancer           12.504505
Asthma           12.446847
Heart Disease    12.432432
Alzheimer’s      12.362162
Infections       12.335135
Name: proportion, dtype: float64
--------------------------------------------------
Doctor
Michael Smith      0.048649
John Smith         0.039640
Robert Smith       0.039640
Michael Johnson    0.036036
James Smith        0.036036
                     ...   
Keith Ortiz        0.001802
Timothy Chapman    0.001802
Terri Collins      0.001802
Jill Ja

# **Findings**

No missing values or duplicate records were identified. The numerical attributes contain valid observations, while categorical attributes contain identifiable and consistent categories.

# **Data Cleaning and Validation**

In [17]:
#Converting admission date from Object format to datetime format
data["Date of Admission"] = pd.to_datetime(data["Date of Admission"])

# Convert discharge date from Object format to datetime format
data["Discharge Date"] = pd.to_datetime(data["Discharge Date"])

In [18]:
print(data[["Date of Admission","Discharge Date"]].dtypes)

Date of Admission    datetime64[ns]
Discharge Date       datetime64[ns]
dtype: object


In [19]:
data[["Date of Admission","Discharge Date"]].sample(2)

,Date of Admission,Discharge Date
19025,2021-03-31,2021-04-08
49124,2021-10-07,2021-11-13


In [20]:
#Checking whether every discharge date occurs after the admission date
Date_Validation = data['Discharge Date'] > data['Date of Admission']
print("Are all discharge date valid: ", Date_Validation.all())

#Checking the number of invalid date records
print("Total Invalid rows in Date_Validation: ",(~Date_Validation).sum())

Are all discharge date valid:  True
Total Invalid rows in Date_Validation:  0


In [21]:
#Calculating Length of Stay independently using admission and discharge dates
Calculated_Length_Of_Stay = (data['Discharge Date'] - data['Date of Admission']).dt.days

#Comparing the calculated values with the existing Length of Stay column
Length_Of_Stay_Validation = data['Length of Stay']== Calculated_Length_Of_Stay
print("Does Length of Stay match the Calculated Length_Of_Stay numbers: ", Length_Of_Stay_Validation.all())

#Checking the number of mismatched records
print("Total Invalid rows in Length_of_Stay_Validation: ",(~Length_Of_Stay_Validation).sum())

Does Length of Stay match the Calculated Length_Of_Stay numbers:  True
Total Invalid rows in Length_of_Stay_Validation:  0


In [22]:
#Checking whether all billing amounts are greater than zero
Billing_Validation = data['Billing Amount']>0
print("Does Billing Amount is greater than zero: ", Billing_Validation.all())

#Checking the invalid billing records
print("Total Invalid rows in Billing Amount: ",(~Billing_Validation).sum())

Does Billing Amount is greater than zero:  True
Total Invalid rows in Billing Amount:  0


In [23]:
#Checking whether patient ages falls within a reasonable range
Age_Validation = data['Age'].between(0,100)
print("Are all ages within the valid ranges: ", Age_Validation.all())

#Count invalid age records
print("Total Invalid Values in Ages: ",(~Age_Validation).sum())

Are all ages within the valid ranges:  True
Total Invalid Values in Ages:  0


In [24]:
# Display the unique values for each categorical Target columns
Categorical_Target_Columns = ["Gender","Blood Type", "Medical Condition", "Hospital", "Insurance Provider", "Admission Type"]

for categorical_columns in Categorical_Target_Columns:
    print(categorical_columns)
    print(data[categorical_columns].unique())
    print("-"*50)

Gender
['Female' 'Male']
--------------------------------------------------
Blood Type
['AB+' 'B-' 'A+' 'O+' 'O-' 'AB-' 'A-' 'B+']
--------------------------------------------------
Medical Condition
['Infections' 'Flu' 'Cancer' 'Asthma' 'Heart Disease' 'Alzheimer’s'
 'Diabetes' 'Obesity']
--------------------------------------------------
Hospital
['Northwestern Memorial Hospital'
 'UI Health (University of Illinois Hospital)' 'UChicago Medicine'
 'Loyola University Medical Center']
--------------------------------------------------
Insurance Provider
['Blue Cross' 'UnitedHealthcare' 'Aetna' 'Cigna' 'Medicare']
--------------------------------------------------
Admission Type
['Emergency' 'Elective' 'Routine' 'Urgent']
--------------------------------------------------


# **Findings**

All discharge dates occur after admission dates, Length of Stay values match the calculated date difference, billing amounts are positive, ages fall within the defined valid range, and categorical values are consistent. Therefore, no rows required deletion due to data-quality violations.

# **Feature Engineering**

In [25]:
#Extracting the day, month and year from the Admission Date
data['Admission Day'] = data['Date of Admission'].dt.day_name()
data['Admission Month'] = data['Date of Admission'].dt.month
data['Admission Year'] = data['Date of Admission'].dt.year

In [26]:
print("Unique values in Admission Year")
print(data["Admission Year"].unique())

Unique values in Admission Year
[2024 2019 2022 2020 2023 2021]


In [27]:
print("Unique values in Admission Week Day")
print(data["Admission Day"].unique())

Unique values in Admission Week Day
['Wednesday' 'Tuesday' 'Thursday' 'Monday' 'Sunday' 'Friday' 'Saturday']


In [28]:
print("Unique values in Admission Month")
print(data["Admission Month"].unique())

Unique values in Admission Month
[ 1  8  9 11 12  7  5  4 10  6  3  2]


In [29]:
print("Frequency values in Admission Year", '\n')
print(data.groupby("Admission Year").size().sort_values(ascending=False))

Frequency values in Admission Year 

Admission Year
2020    11285
2023    11026
2022    11017
2021    10931
2019     7387
2024     3854
dtype: int64


In [30]:
print("Frequency values in Admission Month", '\n')
print(data.groupby("Admission Month").size().sort_values(ascending=False))

Frequency values in Admission Month 

Admission Month
8     4832
7     4812
6     4699
1     4692
10    4678
3     4672
12    4649
5     4599
11    4548
9     4546
4     4518
2     4255
dtype: int64


In [31]:
print("Frequency values in Admission Week Day", '\n')
print(data.groupby("Admission Day").size().sort_values(ascending=False))

Frequency values in Admission Week Day 

Admission Day
Thursday     7989
Tuesday      7982
Wednesday    7950
Sunday       7920
Saturday     7901
Friday       7892
Monday       7866
dtype: int64


# **Findings**

The extracted Temporal features were extracted from the admission date to capture possible historical patterns in admission behavior.
These features were retained as potential predictors for the classification model.

#**One-Hot Encoding**

In [ ]:
# Converting selected categorical variables into numerical dummy variables
Encoded_Data=pd.get_dummies(data, columns=["Gender","Blood Type","Medical Condition","Hospital","Insurance Provider","Admission Day"],drop_first=True,dtype=int)
Encoded_Data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 55500 entries, 0 to 55499
Data columns (total 41 columns):
 #   Column                                                Non-Null Count  Dtype         
---  ------                                                --------------  -----         
 0   Name                                                  55500 non-null  object        
 1   Age                                                   55500 non-null  int64         
 2   Date of Admission                                     55500 non-null  datetime64[ns]
 3   Doctor                                                55500 non-null  object        
 4   Billing Amount                                        55500 non-null  float64       
 5   Room Number                                           55500 non-null  int64         
 6   Admission Type                                        55500 non-null  object        
 7   Discharge Date                                        55500 non-null  dateti

In [ ]:
print(Encoded_Data.columns)

Index(['Name', 'Age', 'Date of Admission', 'Doctor', 'Billing Amount',
       'Room Number', 'Admission Type', 'Discharge Date', 'Medication',
       'Test Results', 'Length of Stay', 'Admission Month', 'Admission Year',
       'Gender_Male', 'Blood Type_A-', 'Blood Type_AB+', 'Blood Type_AB-',
       'Blood Type_B+', 'Blood Type_B-', 'Blood Type_O+', 'Blood Type_O-',
       'Medical Condition_Asthma', 'Medical Condition_Cancer',
       'Medical Condition_Diabetes', 'Medical Condition_Flu',
       'Medical Condition_Heart Disease', 'Medical Condition_Infections',
       'Medical Condition_Obesity', 'Hospital_Northwestern Memorial Hospital',
       'Hospital_UChicago Medicine',
       'Hospital_UI Health (University of Illinois Hospital)',
       'Insurance Provider_Blue Cross', 'Insurance Provider_Cigna',
       'Insurance Provider_Medicare', 'Insurance Provider_UnitedHealthcare',
       'Admission Day_Monday', 'Admission Day_Saturday',
       'Admission Day_Sunday', 'Admission Day_Thu

# **Explanation**

1.   Categorical predictor variables were converted into numerical binary features using one-hot encoding.
2.   drop_first=True was used to avoid redundant dummy variables.
1.   After encoding, the dataset contains 41 columns, including the original
non-feature columns.



# **Target Feature Encoding**

In [ ]:
#Defining the numerical values for each Admission Type category i,e LabelEncoder
Admission_Type_Mapping = {'Emergency': 0, 'Elective': 1, 'Urgent': 2,'Routine':3}

#By using the mapping function converting the Admission Type into numerical values
Encoded_Data['Admission Type'] = Encoded_Data['Admission Type'].map(Admission_Type_Mapping)

In [ ]:
Encoded_Data['Admission Type'].value_counts()

,count
Admission Type,
0,20104
1,13280
2,11816
3,10300


In [ ]:
Encoded_Data.sample(2)

,Name,Age,Date of Admission,Doctor,Billing Amount,Room Number,Admission Type,Discharge Date,Medication,Test Results,...,Insurance Provider_Blue Cross,Insurance Provider_Cigna,Insurance Provider_Medicare,Insurance Provider_UnitedHealthcare,Admission Day_Monday,Admission Day_Saturday,Admission Day_Sunday,Admission Day_Thursday,Admission Day_Tuesday,Admission Day_Wednesday
25605,Mario Gregory,52,2023-12-13,Sherry Oneal,10140.147386,436,3,2023-12-19,Phentermine,Normal,...,0,0,0,0,0,0,0,0,0,1
24074,Thomas Church,24,2021-02-07,Michael Rivera,7706.866270,334,0,2021-02-11,Montelukast,Normal,...,0,0,0,1,0,0,1,0,0,0


# **Explanation**



1.  The target variable Admission Type was converted into numerical class labels i,e Emergency=0, Elective=1, Urgent=2 and Routine=3.
2. The original class distribution is retained after encoding.



# **Model Selection**

In [ ]:
#Importing train_test_split to divide the dataset into training and testing data
from sklearn.model_selection import train_test_split

#Importing classification models
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier, GradientBoostingClassifier

# Import evaluation metrics for classification performance
from sklearn.metrics import classification_report, confusion_matrix
from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score  #Model Evaluation Parameters

In [ ]:
#Splitting the Training and Target Features
x=Encoded_Data.drop(['Name','Date of Admission', 'Doctor','Billing Amount','Room Number','Admission Type','Discharge Date', 'Medication','Test Results', 'Length of Stay'], axis=1)
y=Encoded_Data['Admission Type']

In [ ]:
print(x.shape,y.shape)

(55500, 31) (55500,)


# **Explanation**

1. Features that were identifiers, highly high-cardinality attributes, post-admission information, or variables potentially unavailable at the time of admission-type prediction were excluded from the primary classification model

2. This was done to reduce noise and avoid potential data leakage.

# **Train and Test Split**

In [ ]:
X_train, X_test, Y_train, Y_test = train_test_split(x,y, test_size=0.2, random_state=2, stratify=y)

In [ ]:
print(X_train.shape, X_test.shape, Y_train.shape, Y_test.shape)

(44400, 31) (11100, 31) (44400,) (11100,)


# **Explanation**

1. The dataset was divided into 80% training data and 20% testing data.
2. Stratified splitting was used to preserve the distribution of the four admission-type classes in both subsets i,e preserves the
target-class distribution.

# **Baseline Model Selection**

In [ ]:
Models={
'Logistic Regression':LogisticRegression(max_iter=3000,solver='saga'),
'Decision Tree':DecisionTreeClassifier(),
'Random Forest':RandomForestClassifier(random_state=2)
 }


# **Explanation**
1. Logistic Regression → Provides a simple linear classification baseline.

2. Decision Tree → Captures non-linear decision patterns and is easy to interpret.

3. Random Forest → Combines multiple decision trees and can capture more complex patterns while providing feature importance.

In [ ]:
#Creating an Empty Result Set
Results = []

In [ ]:
#Train and evaluate each baseline model
for model_name, model in Models.items():

      #Training the model using the training dataset
      model.fit(X_train, Y_train)

      #Generate predictions for the unseen test dataset
      Y_pred = model.predict(X_test)

      #Calculating classification metrics
      accuracy = accuracy_score(Y_test, Y_pred)*100
      precision = precision_score(Y_test, Y_pred, average='weighted',zero_division=0)*100
      recall = recall_score(Y_test, Y_pred, average='weighted',zero_division=0)*100
      f1 = f1_score(Y_test, Y_pred, average='weighted',zero_division=0)*100

      Results.append([model_name, accuracy,precision,recall,f1])


In [ ]:
#Convert the results into a DataFrame
Results_df = pd.DataFrame(Results, columns=['Model', 'Accuracy', 'Precision', 'Recall', 'F1-Score'])
Results_df

,Model,Accuracy,Precision,Recall,F1-Score
0,Logistic Regression,45.306306,27.580436,45.306306,33.771036
1,Decision Tree,33.216216,33.555208,33.216216,33.370806
2,Random Forest,41.225225,36.872034,41.225225,37.537292


# **Findings**

1. The baseline models show that the classification problem is challenging.
2. Logistic Regression gives the highest baseline accuracy.
3. Random Forest gives the highest baseline weighted F1 score.
4. Decision Tree performs the weakest in terms of accuracy.

# **Classification Report and Confusion Matrix**

In [ ]:
#Displaying detailed classification performance for each baseline model
for model_name, model in Models.items():
  model.fit(X_train, Y_train)

  #Generating predictions using the trained model
  Y_pred = model.predict(X_test)
  print(f"Classification Report for {model_name}:")

  #Displaying precision, recall and F1 score for each class
  print(classification_report(Y_test, Y_pred))

  #Displaying the confusion matrix
  print(f"Confusion Matrix for {model_name}:")
  print(confusion_matrix(Y_test, Y_pred))
  print("-"*70)

Classification Report for Logistic Regression:
              precision    recall  f1-score   support

           0       0.44      0.87      0.59      4021
           1       0.49      0.57      0.52      2656
           2       0.00      0.00      0.00      2363
           3       0.00      0.00      0.00      2060

    accuracy                           0.45     11100
   macro avg       0.23      0.36      0.28     11100
weighted avg       0.28      0.45      0.34     11100

Confusion Matrix for Logistic Regression:
[[3515  505    0    1]
 [1142 1514    0    0]
 [1874  488    0    1]
 [1450  609    1    0]]
----------------------------------------------------------------------
Classification Report for Decision Tree:
              precision    recall  f1-score   support

           0       0.44      0.41      0.42      4021
           1       0.35      0.35      0.35      2656
           2       0.23      0.25      0.24      2363
           3       0.22      0.22      0.22      2060


# **Findings**

The classification reports reveal that the model performs substantially better for Emergency and Elective than for Urgent and Routine.

# **Feature Importance**

In [ ]:
#Evaluating the Importance of Features by using the Random Forest Model from the Trained Model
Random_Forest_model = Models['Random Forest']

#Extract the importance of every feature
importances = (Random_Forest_model.feature_importances_)*100
feature_importances = pd.DataFrame({'Feature': x.columns, 'Importance': importances}).sort_values(by='Importance', ascending=False)
print(feature_importances)

                                              Feature  Importance
0                                                 Age   21.620510
1                                     Admission Month   15.269335
2                                      Admission Year   11.005430
3                                         Gender_Male    3.898197
17                          Medical Condition_Obesity    3.560677
9                                       Blood Type_O+    2.882941
18            Hospital_Northwestern Memorial Hospital    2.676961
19                         Hospital_UChicago Medicine    2.583594
20  Hospital_UI Health (University of Illinois Hos...    2.433679
21                      Insurance Provider_Blue Cross    2.367574
23                        Insurance Provider_Medicare    2.213449
24                Insurance Provider_UnitedHealthcare    2.199929
22                           Insurance Provider_Cigna    2.199357
29                              Admission Day_Tuesday    1.983042
30        

# **Explanation**

1. Age, Admission Month and Admission Year have the highest relative feature importance in the baseline Random Forest model.

2. These values indicate the model's relative reliance on the features during tree-based splitting and should not be interpreted as causal relationships.

# **Random Forest Hyperparameter Evaluation**

In [ ]:
#Defining multiple Random Forest configurations

Random_Forest_Models ={
    'RF_L1': RandomForestClassifier(n_estimators=300,random_state=2,max_depth=10,min_samples_split=10,min_samples_leaf=1),  # Configuration 1
    'RF_L2': RandomForestClassifier(n_estimators=300,random_state=2,max_depth=10,min_samples_split=10,min_samples_leaf=2),  # Configuration 2
    'RF_L3': RandomForestClassifier(n_estimators=300,random_state=2,max_depth=10,min_samples_split=10,min_samples_leaf=5),  # Configuration 3
    'RF_L4': RandomForestClassifier(n_estimators=300,random_state=2,max_depth=20,min_samples_split=20,min_samples_leaf=10)  # Configuration 4
    }

#Storing Random Forest tuning results
Random_Results=[]

#Train and evaluate each Random Forest configuration
for model_name, model in Random_Forest_Models.items():
      model.fit(X_train, Y_train)
      Y_pred = model.predict(X_test)
      accuracy = accuracy_score(Y_test, Y_pred)*100
      precision = precision_score(Y_test, Y_pred, average='weighted',zero_division=0)*100
      recall = recall_score(Y_test, Y_pred, average='weighted',zero_division=0)*100
      f1 = f1_score(Y_test, Y_pred, average='weighted',zero_division=0)*100
      Random_Results.append([model_name, accuracy,precision,recall,f1])

#Convert results into a DataFrame
Results_Random_df = pd.DataFrame(Random_Results, columns=['Model', 'Accuracy','precision','recall','f1'])
Results_Random_df

,Model,Accuracy,precision,recall,f1
0,RF_L1,45.306306,27.668062,45.306306,33.721671
1,RF_L2,45.288288,27.693278,45.288288,33.694089
2,RF_L3,45.297297,27.729053,45.297297,33.677094
3,RF_L4,45.567568,40.450176,45.567568,36.210198


In [ ]:
#Displaying detailed classification performance for each Random Forest model
for model_name, model in Random_Forest_Models.items():
  model.fit(X_train, Y_train)

  #Generating predictions using the trained model
  Y_pred = model.predict(X_test)
  print(f"Classification Report for {model_name}:")

  #Displaying precision, recall and F1 score for each class
  print(classification_report(Y_test, Y_pred, zero_division=0))

  #Displaying the confusion matrix
  print(f"Confusion Matrix for {model_name}:")
  print(confusion_matrix(Y_test, Y_pred))
  print("-"*70)

Classification Report for RF_L1:
              precision    recall  f1-score   support

           0       0.44      0.88      0.59      4021
           1       0.49      0.55      0.52      2656
           2       0.00      0.00      0.00      2363
           3       0.00      0.00      0.00      2060

    accuracy                           0.45     11100
   macro avg       0.23      0.36      0.28     11100
weighted avg       0.28      0.45      0.34     11100

Confusion Matrix for RF_L1:
[[3557  464    0    0]
 [1184 1472    0    0]
 [1897  466    0    0]
 [1470  589    1    0]]
----------------------------------------------------------------------
Classification Report for RF_L2:
              precision    recall  f1-score   support

           0       0.44      0.89      0.59      4021
           1       0.49      0.55      0.52      2656
           2       0.00      0.00      0.00      2363
           3       0.00      0.00      0.00      2060

    accuracy                       

# **Findings**

Among the evaluated Random Forest configurations, RF_L4 achieved the highest accuracy and weighted precision. It also improved the weighted F1-score compared with the other tested configurations. Therefore, RF_L4 was selected as the final Random Forest configuration.

# **Gradient Boosting Evaluation**

In [ ]:
#Defining multiple Gradient Boosting configurations
Gradient_Models ={
    'GB_L1': GradientBoostingClassifier(n_estimators=300,learning_rate=0.01,random_state=2), # Lower learning rate
    'GB_L2': GradientBoostingClassifier(n_estimators=300,learning_rate=0.05,random_state=2), # Moderate learning rate
    'GB_L3': GradientBoostingClassifier(n_estimators=300,learning_rate=0.10,random_state=2), # Higher learning rate
    'GB_L4': GradientBoostingClassifier(n_estimators=300,learning_rate=0.20,random_state=2)  # Highest Tested learning rate
    }

#Storing Gradient Boosting evaluation results
Gradient_Models_Result=[]

#Train and evaluate each Gradient Boosting configuration
for model_name, model in Gradient_Models.items():
      model.fit(X_train, Y_train)
      Y_pred = model.predict(X_test)
      accuracy = accuracy_score(Y_test, Y_pred)*100
      precision = precision_score(Y_test, Y_pred, average='weighted',zero_division=0)*100
      recall = recall_score(Y_test, Y_pred, average='weighted',zero_division=0)*100
      f1 = f1_score(Y_test, Y_pred, average='weighted',zero_division=0)*100
      Gradient_Models_Result.append([model_name, accuracy,precision,recall,f1])

#Convert results into a DataFrame
Gradient_Models_df = pd.DataFrame(Gradient_Models_Result, columns=['Model', 'Accuracy','precision','recall','f1'])
Gradient_Models_df

,Model,Accuracy,precision,recall,f1
0,GB_L1,45.279279,32.696856,45.279279,33.993149
1,GB_L2,45.126126,37.812460,45.126126,35.490544
2,GB_L3,45.369369,39.087591,45.369369,36.630134
3,GB_L4,44.882883,38.336313,44.882883,37.179380


In [ ]:
#Displaying detailed classification performance for each Gradient Boosting model
for model_name, model in Gradient_Models.items():
  model.fit(X_train, Y_train)

  #Generating predictions using the trained model
  Y_pred = model.predict(X_test)
  print(f"Classification Report for {model_name}:")

  #Displaying precision, recall and F1 score for each class
  print(classification_report(Y_test, Y_pred,zero_division=0))

  #Displaying the confusion matrix
  print(f"Confusion Matrix for {model_name}:")
  print(confusion_matrix(Y_test, Y_pred))
  print("-"*70)

Classification Report for GB_L1:
              precision    recall  f1-score   support

           0       0.44      0.86      0.58      4021
           1       0.48      0.59      0.53      2656
           2       0.00      0.00      0.00      2363
           3       0.28      0.01      0.01      2060

    accuracy                           0.45     11100
   macro avg       0.30      0.36      0.28     11100
weighted avg       0.33      0.45      0.34     11100

Confusion Matrix for GB_L1:
[[3445  561    0   15]
 [1082 1570    0    4]
 [1826  528    0    9]
 [1407  642    0   11]]
----------------------------------------------------------------------
Classification Report for GB_L2:
              precision    recall  f1-score   support

           0       0.46      0.81      0.58      4021
           1       0.46      0.62      0.53      2656
           2       0.24      0.00      0.01      2363
           3       0.28      0.05      0.08      2060

    accuracy                       

# **Findings**

1. Gradient Boosting produced comparable performance to Random Forest.
2. GB_L3 achieved the highest accuracy and precision among the Gradient Boosting configurations, while GB_L4 achieved the highest weighted F1-score within the Gradient Boosting experiments.

## **ML Conclusion**

Multiple classification algorithms and hyperparameter configurations were evaluated. The final Random Forest configuration, RF_L4, achieved the highest observed accuracy of approximately 45.57% and weighted precision of approximately 40.45%. Although GB_L4 achieved a slightly higher weighted F1-score of approximately 37.18%, RF_L4 was selected as the final model because of its higher accuracy and precision and its suitability for feature-importance analysis within the proposed resource-optimization framework.

Class-wise evaluation showed that the model performs considerably better for Emergency and Elective admissions than for Urgent and Routine admissions. Therefore, the model is used as a historical decision-support component rather than a clinically validated admission decision system.

# **SQL Analytics**

In [36]:
#Importing the Sqlite3 library
import sqlite3 as sql

In [37]:
print(sql.sqlite_version)

3.45.1


In [38]:
connection = sql.connect('Healthcare_Data.db')
print("Database created and Successfully Connected to SQLite")

Database created and Successfully Connected to SQLite


In [40]:
#Loading the Cleaned dataset into sql
data.to_sql('Healthcare_Data', connection, if_exists='replace', index=False)
print("Dataframe is successfully stored into a database")

Dataframe is successfully stored into a database


In [41]:
#Verifiying tables exists or not
Tables=pd.read_sql_query("SELECT name FROM sqlite_master WHERE type='table'", connection)
print(Tables)

              name
0  Healthcare_Data


In [44]:
#Cross checking the Total Rows and Columns in the table
df = pd.read_sql_query("SELECT * FROM Healthcare_Data", connection)
print(f'Total Rows: {df.shape[0]}')
print(f'Total Columns: {df.shape[1]}')

Total Rows: 55500
Total Columns: 19


In [47]:
#Displaying the Top 2 rows of the Dataset
Top_2_Rows = pd.read_sql_query("SELECT * FROM Healthcare_Data LIMIT 2", connection)
Top_2_Rows

,Name,Age,Gender,Blood Type,Medical Condition,Date of Admission,Doctor,Hospital,Insurance Provider,Billing Amount,Room Number,Admission Type,Discharge Date,Medication,Test Results,Length of Stay,Admission Day,Admission Month,Admission Year
0,Bobby Jackson,19,Female,AB+,Infections,2024-01-31 00:00:00,Matthew Smith,Northwestern Memorial Hospital,Blue Cross,2212.272701,328,Emergency,2024-02-07 00:00:00,Azithromycin,Normal,7,Wednesday,1,2024
1,Leslie Terry,15,Female,B-,Flu,2019-08-20 00:00:00,Samantha Davies,UI Health (University of Illinois Hospital),UnitedHealthcare,3185.161388,265,Emergency,2019-08-22 00:00:00,Tamiflu,Abnormal,2,Tuesday,8,2019


In [71]:
#Distribution of Patients across different Admission Types

Query = '''SELECT
                 [Admission Type],
                 Count(*) as 'Patients'
           FROM Healthcare_Data
           GROUP BY [Admission Type]
           ORDER BY Patients DESC'''

Admission_Type_Distribution = pd.read_sql_query(Query,connection)

Admission_Type_Distribution

,Admission Type,Patients
0,Emergency,20104
1,Elective,13280
2,Urgent,11816
3,Routine,10300


In [72]:
#Distribution of Patients across different Admission Types along with Percentage
query='''SELECT
             [Admission Type],
             Count(*) AS 'Patients',
             Count(*) * 100.0/(select count(*) from Healthcare_Data) AS Percentage
          FROM Healthcare_Data
          GROUP BY [Admission Type]
          ORDER BY Patients desc'''


Admission_Type_Percentage = pd.read_sql_query(query,connection)

Admission_Type_Percentage

,Admission Type,Patients,Percentage
0,Emergency,20104,36.223423
1,Elective,13280,23.927928
2,Urgent,11816,21.290090
3,Routine,10300,18.558559


In [73]:
#Hospital-Wise Patient Workload
query='''SELECT
                Hospital,
                Count(*) AS 'Patients'
          FROM Healthcare_Data
          GROUP BY Hospital
          ORDER BY Patients DESC'''


Hospital_Wise_Patient_Workload = pd.read_sql_query(query,connection)

Hospital_Wise_Patient_Workload

,Hospital,Patients
0,Northwestern Memorial Hospital,14070
1,UI Health (University of Illinois Hospital),13950
2,UChicago Medicine,13870
3,Loyola University Medical Center,13610


In [74]:
#Hospital-Wise Patient Average Stay
query='''SELECT
                Hospital,
                Avg([Length of Stay]) AS 'Average_LOS'
          FROM Healthcare_Data
          GROUP BY Hospital
          ORDER BY Average_LOS DESC'''


Hospital_Wise_Patient_Average_LOS = pd.read_sql_query(query,connection)

Hospital_Wise_Patient_Average_LOS

,Hospital,Average_LOS
0,Northwestern Memorial Hospital,17.907321
1,UI Health (University of Illinois Hospital),17.894409
2,Loyola University Medical Center,17.786995
3,UChicago Medicine,17.773035


In [75]:
#Hospital Workload Analysis
query='''SELECT
                Hospital,
                count(*) AS 'Patients',
                Avg([Length of Stay]) AS 'Average_LOS'
          FROM Healthcare_Data
          GROUP BY Hospital
          ORDER BY Patients DESC, Average_LOS DESC'''


Hospital_Workload_Analysis = pd.read_sql_query(query,connection)

Hospital_Workload_Analysis

,Hospital,Patients,Average_LOS
0,Northwestern Memorial Hospital,14070,17.907321
1,UI Health (University of Illinois Hospital),13950,17.894409
2,UChicago Medicine,13870,17.773035
3,Loyola University Medical Center,13610,17.786995


In [80]:
#Hospital Wise Admission Type Analysis
query='''SELECT
                Hospital,
                [Admission Type],
                count(*) AS 'Patients'
          FROM Healthcare_Data
          GROUP BY Hospital, [Admission Type]
          ORDER BY Hospital, Patients DESC'''


Hospital_Admission_Analysis = pd.read_sql_query(query,connection)

Hospital_Admission_Analysis

,Hospital,Admission Type,Patients
0,Loyola University Medical Center,Emergency,4936
1,Loyola University Medical Center,Elective,3210
2,Loyola University Medical Center,Urgent,2900
3,Loyola University Medical Center,Routine,2564
4,Northwestern Memorial Hospital,Emergency,5122
5,Northwestern Memorial Hospital,Elective,3367
6,Northwestern Memorial Hospital,Urgent,2981
7,Northwestern Memorial Hospital,Routine,2600
8,UChicago Medicine,Emergency,5042
9,UChicago Medicine,Elective,3319


In [81]:
#Hospital Wise Admission Type Percentage
query='''SELECT
                Hospital,
                [Admission Type],
                count(*) AS 'Total_Patients',
                sum(count(*)) OVER (PARTITION BY Hospital) AS 'Total_Hospital_Counts',
                count(*) * 100.0 / sum(count(*)) OVER (PARTITION BY Hospital) AS 'Percentage'
          FROM Healthcare_Data
          GROUP BY Hospital, [Admission Type]
          ORDER BY Hospital, Percentage DESC'''


Hospital_Admission_Percentage = pd.read_sql_query(query,connection)

Hospital_Admission_Percentage

,Hospital,Admission Type,Total_Patients,Total_Hospital_Counts,Percentage
0,Loyola University Medical Center,Emergency,4936,13610,36.267450
1,Loyola University Medical Center,Elective,3210,13610,23.585599
2,Loyola University Medical Center,Urgent,2900,13610,21.307862
3,Loyola University Medical Center,Routine,2564,13610,18.839089
4,Northwestern Memorial Hospital,Emergency,5122,14070,36.403696
5,Northwestern Memorial Hospital,Elective,3367,14070,23.930348
6,Northwestern Memorial Hospital,Urgent,2981,14070,21.186923
7,Northwestern Memorial Hospital,Routine,2600,14070,18.479033
8,UChicago Medicine,Emergency,5042,13870,36.351839
9,UChicago Medicine,Elective,3319,13870,23.929344


In [83]:
#Medical Condition Patient Distribution
query='''SELECT
                [Medical Condition],
                count(*) AS 'Patients'
          FROM Healthcare_Data
          GROUP BY [Medical Condition]
          ORDER BY Patients DESC'''


Medical_Condition_Patient_Distribution = pd.read_sql_query(query,connection)

Medical_Condition_Patient_Distribution

,Medical Condition,Patients
0,Flu,7046
1,Diabetes,7005
2,Obesity,6994
3,Cancer,6940
4,Asthma,6908
5,Heart Disease,6900
6,Alzheimer’s,6861
7,Infections,6846


In [84]:
#Medical Condition Average LOS
query='''SELECT
                [Medical Condition],
                Avg([Length of Stay]) AS 'Average_LOS'
          FROM Healthcare_Data
          GROUP BY [Medical Condition]
          ORDER BY Average_LOS DESC'''


Medical_Condition_Average_LOS = pd.read_sql_query(query,connection)

Medical_Condition_Average_LOS

,Medical Condition,Average_LOS
0,Alzheimer’s,54.417286
1,Cancer,36.543804
2,Heart Disease,26.860580
3,Diabetes,8.063098
4,Obesity,5.966400
5,Infections,5.521911
6,Asthma,3.503619
7,Flu,2.504967


In [115]:
#Medical Condition Workload Analysis
query='''SELECT
                [Medical Condition],
                count(*) AS 'Patients',
                Avg([Length of Stay]) AS 'Average_LOS'
          FROM Healthcare_Data
          GROUP BY [Medical Condition]
          ORDER BY Patients DESC, Average_LOS DESC'''


Medical_Condition_Workload_Analysis = pd.read_sql_query(query,connection)

Medical_Condition_Workload_Analysis

,Medical Condition,Patients,Average_LOS
0,Flu,7046,2.504967
1,Diabetes,7005,8.063098
2,Obesity,6994,5.966400
3,Cancer,6940,36.543804
4,Asthma,6908,3.503619
5,Heart Disease,6900,26.860580
6,Alzheimer’s,6861,54.417286
7,Infections,6846,5.521911


In [131]:
#Monthly Admission Trend
query='''SELECT
                [Admission Month],
                count(*) AS 'Patients'
          FROM Healthcare_Data
          GROUP BY [Admission Month]
          ORDER BY [Admission Month]'''


Monthly_Admission_Trend = pd.read_sql_query(query,connection)

Monthly_Admission_Trend

,Admission Month,Patients
0,1,4692
1,2,4255
2,3,4672
3,4,4518
4,5,4599
5,6,4699
6,7,4812
7,8,4832
8,9,4546
9,10,4678


In [88]:
#Yearly Admission Trend
query='''SELECT
                [Admission Year],
                count(*) AS 'Patients'
          FROM Healthcare_Data
          GROUP BY [Admission Year]
          ORDER BY [Admission Year]'''


Yearly_Admission_Trend = pd.read_sql_query(query,connection)

Yearly_Admission_Trend

,Admission Year,Patients
0,2019,7387
1,2020,11285
2,2021,10931
3,2022,11017
4,2023,11026
5,2024,3854


In [90]:
#Top 10 Doctor Patient Workload
query='''SELECT
                Doctor,
                count(*) AS 'Patients'
          FROM Healthcare_Data
          GROUP BY Doctor
          ORDER BY Patients DESC LIMIT 10'''

Top_10_Doctor_Patient_Workload = pd.read_sql_query(query,connection)

Top_10_Doctor_Patient_Workload

,Doctor,Patients
0,Michael Smith,27
1,Robert Smith,22
2,John Smith,22
3,Michael Johnson,20
4,James Smith,20
5,Robert Johnson,19
6,David Smith,19
7,Michael Williams,18
8,Matthew Smith,17
9,John Johnson,17


In [101]:
#Total Billing by Hospital
query='''SELECT
                Hospital,
                CAST(sum([Billing Amount]) AS BIGINT) AS 'Total_Billing'
          FROM Healthcare_Data
          GROUP BY Hospital
          ORDER BY Total_Billing DESC'''

Hospital_Wise_Total_Billing = pd.read_sql_query(query,connection)

Hospital_Wise_Total_Billing

,Hospital,Total_Billing
0,Northwestern Memorial Hospital,308689213
1,UI Health (University of Illinois Hospital),303837582
2,UChicago Medicine,301281681
3,Loyola University Medical Center,298036500


In [103]:
#Average Billing by Hospital
query='''SELECT
                Hospital,
                CAST(Avg([Billing Amount]) AS BIGINT) AS 'Average_Billing'
          FROM Healthcare_Data
          GROUP BY Hospital
          ORDER BY Average_Billing DESC'''

Hospital_Wise_Average_Billing = pd.read_sql_query(query,connection)

Hospital_Wise_Average_Billing

,Hospital,Average_Billing
0,Northwestern Memorial Hospital,21939
1,Loyola University Medical Center,21898
2,UI Health (University of Illinois Hospital),21780
3,UChicago Medicine,21721


In [116]:
#Hospital Resource Demand Analysis
query='''SELECT
                Hospital,
                Count(*) AS 'Patients',
                sum(CASE
                          WHEN [Admission Type]="Emergency"
                          THEN 1
                          ELSE 0
                    END) AS Emergency_Patients,

                sum(CASE
                          WHEN [Admission Type]="Emergency"
                          THEN 1
                          ELSE 0
                    END) *100.0 / count(*) AS Emergency_Percentage,

                sum(CASE
                          WHEN [Admission Type]="Elective"
                          THEN 1
                          ELSE 0
                    END) AS Elective_Patients,

                sum(CASE
                          WHEN [Admission Type]="Elective"
                          THEN 1
                          ELSE 0
                    END) *100.0 / count(*) AS Elective_Percentage,


                sum(CASE
                          WHEN [Admission Type]="Urgent"
                          THEN 1
                          ELSE 0
                    END) AS Urgent_Patients,

                sum(CASE
                          WHEN [Admission Type]="Urgent"
                          THEN 1
                          ELSE 0
                    END) *100.0 / count(*) AS Urgent_Percentage,

                sum(CASE
                          WHEN [Admission Type]="Routine"
                          THEN 1
                          ELSE 0
                    END) AS Routine_Patients,
                 sum(CASE
                          WHEN [Admission Type]="Routine"
                          THEN 1
                          ELSE 0
                    END) *100.0 / count(*) AS Routine_Percentage,

                Avg([Length of Stay]) AS 'Average_LOS'
          FROM Healthcare_Data
          GROUP BY Hospital
          ORDER BY Patients DESC, Emergency_Patients DESC, Average_LOS DESC'''

Hospital_Resource_Demand_Analysis = pd.read_sql_query(query,connection)

Hospital_Resource_Demand_Analysis

,Hospital,Patients,Emergency_Patients,Emergency_Percentage,Elective_Patients,Elective_Percentage,Urgent_Patients,Urgent_Percentage,Routine_Patients,Routine_Percentage,Average_LOS
0,Northwestern Memorial Hospital,14070,5122,36.403696,3367,23.930348,2981,21.186923,2600,18.479033,17.907321
1,UI Health (University of Illinois Hospital),13950,5004,35.870968,3384,24.258065,3013,21.598566,2549,18.272401,17.894409
2,UChicago Medicine,13870,5042,36.351839,3319,23.929344,2922,21.067051,2587,18.651766,17.773035
3,Loyola University Medical Center,13610,4936,36.267450,3210,23.585599,2900,21.307862,2564,18.839089,17.786995


# **Data Visualization**

In [117]:
#Importing the Plotly  library
import plotly.express as px

In [194]:
#Admission Type Distribution

Admission_Type_Color_Mapping = {'Emergency':"#4E79A7", 'Elective':"#F28E2B", 'Urgent':"#E15759", 'Routine':"#76B7B2"}

fig = px.bar(
    Admission_Type_Distribution,
    x="Admission Type",
    y="Patients",
    title="Admission Type Distribution",
    text="Patients",
    color="Admission Type",
    color_discrete_map=Admission_Type_Color_Mapping)

fig.update_layout(
    title={'x':0.5,'xanchor':"center"},
    xaxis_title="Admission Type",
    yaxis_title="Number of Patients",
    showlegend=False
)

fig.show()

In [155]:
#Admission Month Trend

fig = px.line(
    Monthly_Admission_Trend,
    x="Admission Month",
    y="Patients",
    title="Monthly Admission Trend Over Time",
    markers=True,
    line_shape="spline")

fig.update_layout(
    title={'x':0.5,'xanchor':"center"},
    xaxis_title="Month",
    yaxis_title="Number of Admissions",
     xaxis=
        dict(
              tickmode="array",
              tickvals=list(range(1, 13)),
              ticktext=[
                  "Jan", "Feb", "Mar", "Apr", "May", "Jun",
                  "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"
                  ]
            )
)

fig.update_traces(
    line=dict(color="#2ca02c", width=3),
    marker=dict(size=10, color="#2ca02c")
)

fig.show()

In [168]:
#Admission Year Trend

fig = px.line(
    Yearly_Admission_Trend,
    x="Admission Year",
    y="Patients",
    title="Yearly Admission Trend (2019-2024)",
    markers=True,
    line_shape="vhv")

fig.update_layout(
                    title={'x':0.5,'xanchor':"center"},
                    xaxis_title="Year",
                    yaxis_title="Number of Patients"
                 )

fig.update_traces(
                  line=dict(color="#E65100", width=3),
                  marker=dict(size=10, color="#FF6B6B")
                )

fig.show()

In [206]:
#Hospital Wise Patient Workload

Hospital_Color_Mapping = {'Northwestern Memorial Hospital':"#00897B",
                          'UI Health (University of Illinois Hospital)':"#5C6BC0",
                          'UChicago Medicine':"#26A69A",
                          'Loyola University Medical Center':"#AB47BC"}

fig = px.bar(
    Hospital_Wise_Patient_Workload,
    x="Hospital",
    y="Patients",
    title="Hospital Wise Patient Workload",
    text="Patients",
    color="Hospital",
    color_discrete_map=Hospital_Color_Mapping)

fig.update_layout(
    title={'x':0.5,'xanchor':"center"},
    xaxis_title="Hospital",
    yaxis_title="Number of Patients",
    xaxis_tickangle=0,
    showlegend=False
)

fig.show()

In [204]:
#Medical Condition Distribution

Medical_Conditions_Color_Mapping = {
   'Flu':"#2E5BFF",
   'Diabetes':"#008080",
    'Obesity':"#7B1FA2",
    'Cancer':"#C2185B",
    'Asthma':"#E65100",
    'Heart Disease':"#37474F",
    '[Alzheimer’s]':"#0288D1",
    'Infections':"#689F38"
}


fig = px.bar(
    Medical_Condition_Patient_Distribution,
    x="Medical Condition",
    y="Patients",
    title="Medical Condition Distribution",
    text="Patients",
    color="Medical Condition",
    color_discrete_map=Medical_Conditions_Color_Mapping)

fig.update_layout(
    title={'x':0.5,'xanchor':"center"},
    xaxis_title="Medical Conditions",
    yaxis_title="Number of Patients",
    xaxis_tickangle=0,
    showlegend=False
)

fig.show()